### PR TITLE
Prevent premature call to SetApplicationIcon (bnc#940341)

### PR DESCRIPTION
**Pull request by ancorgs**

### CODE REVIEW

File: library/wizard/src/modules/Wizard.rb [hunk 1]
```diff
@@ -1192,9 +1166,8 @@ def SetDesktopIcon(file)
 
       return false unless icon
 
-      icon_path = File.join(@icon_path, icon + ".png")
```
Comment (mvidner) [inline library/wizard/src/modules/Wizard.rb]: So this @icon_path was never defined and wouldn't work. Nice catch.

File: library/wizard/src/modules/Wizard.rb [hunk 2]
```diff
@@ -1881,6 +1851,24 @@ def SetProductName(name)
...
+    # arguments
+    def open_wizard_dialog(*args)
+      set_icon
+      UI.OpenDialog(
+        Opt(:wizardDialog),
+        Wizard(*args)
+      )
+    end
+
+    # Sets the application icon according to the value of @icon_name
```
Comment (mvidner) [inline library/wizard/src/modules/Wizard.rb]: "This should be called only before an actual dialog is opened; premature UI calls can interfere with the CommandLine mode"

Comment (ancorgs) [inline library/wizard/src/modules/Wizard.rb]: Good point

**Comment by mvidner** [comment]
LGTM.
Last time Rubocop complained about the indentation of `protected` BTW.

**Pull request merged**